packages/dev/core/src/Physics/v2/IPhysicsEnginePluginV2.ts

```diff
@@ -0,0 +1,180 @@
+import type { Vector3, Quaternion } from "../../Maths/math.vector";
+import type { AbstractMesh } from "../../Meshes/abstractMesh";
+import type { PhysicsRaycastResult } from "../physicsRaycastResult";
+import type { PhysicsBody } from "./physicsBody";
+import type { PhysicsShape } from "./physicsShape";
+import type { PhysicsConstraint } from "./physicsConstraint";
+import type { BoundingBox } from "../../Culling/boundingBox";
+import type { TransformNode } from "../../Meshes/transformNode";
+import type { PhysicsMaterial } from "./physicsMaterial";
+
+export enum ConstraintAxisLimitMode {
+    FREE,
+    LIMITED,
+    LOCKED,
+    NONE,
+}
+
+export enum ConstraintAxis {
+    LINEAR_X,
+    LINEAR_Y,
+    LINEAR_Z,
+    ANGULAR_X,
+    ANGULAR_Y,
+    ANGULAR_Z,
+    LINEAR_DISTANCE,
+}
+export enum ConstraintType {
+    BALL_AND_SOCKET,
+    DISTANCE,
+    HINGE,
+    SLIDER,
+    LOCK,
+}
+
+export enum ShapeType {
+    SPHERE,
+    CAPSULE,
+    CYLINDER,
+    BOX,
+    CONVEX_HULL,
+    CONTAINER,
+    MESH,
+}
+
+export enum ConstraintMotorType {
+    NONE,
+    VELOCITY,
+    POSITION,
+}
+
+export interface PhysicsShapeParameters {
+    center?: Vector3;
+    radius?: number;
+    pointA?: Vector3;
+    pointB?: Vector3;
+    rotation?: Quaternion;
+    extents?: Vector3;
+    mesh?: AbstractMesh;
+}
+
+export interface PhysicsConstraintParameters {
+    pivotA?: Vector3;
+    pivotB?: Vector3;
+    axisA?: Vector3;
+    axisB?: Vector3;
+}
+
+/**
+ *
+ */
+export interface MassProperties {
+    /**
+     *
+     */
+    centerOfMass: Vector3;
+    /**
+     *
+     */
+    mass: number;
+    /**
+     *
+     */
+    intertia: Vector3;
+    /**
+     *
+     */
+    inertiaOrientation: Quaternion;
+}
+
+/** @internal */
+export interface IPhysicsEnginePluginV2 {
+    /**
+     *
+     */
+    world: any;
+    /**
+     *
+     */
+    name: string;
+    setGravity(gravity: Vector3): void;
+    setTimeStep(timeStep: number): void;
+    getTimeStep(): number;
+    executeStep(delta: number): void; //not forgetting pre and post events
+
+    // body
+    initBody(body: PhysicsBody): void;
+    setShape(body: PhysicsBody, shape: PhysicsShape): void;
+    getShape(body: PhysicsBody): PhysicsShape;
+    setFilterGroup(body: PhysicsBody, group: number): void;
+    getFilterGroup(body: PhysicsBody): number;
+    setEventMask(body: PhysicsBody, eventMask: number): void;
+    getEventMask(body: PhysicsBody): number;
+    setMassProperties(body: PhysicsBody, massProps: MassProperties): void;
+    getMassProperties(body: PhysicsBody): MassProperties;
+    setLinearDamping(body: PhysicsBody, damping: number): void;
+    getLinearDamping(body: PhysicsBody): number;
+    setAngularDamping(body: PhysicsBody, damping: number): void;
+    getAngularDamping(body: PhysicsBody): number;
+    setLinearVelocity(body: PhysicsBody, linVel: Vector3): void;
+    getLinearVelocity(body: PhysicsBody): Vector3;

```

Comment:
maybe we can make those functions ToRef (or add ToRef) so that we don't create a new vector3 on every call?

Reply:
Fixed